pkg/scalers/neo4j_scaler.go

```diff
@@ -0,0 +1,209 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	"github.com/neo4j/neo4j-go-driver/v5/neo4j"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type neo4jScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *neo4jMetadata
+	driver     neo4j.DriverWithContext
+	logger     logr.Logger
+}
+
+type neo4jMetadata struct {
+	connectionString     string
+	host                 string
+	port                 string
+	username             string
+	password             string
+	query                string
+	queryValue           int64
+	activationQueryValue int64
+	metricName           string
+	scalerIndex          int
+}
+
+func (n *neo4jScaler) GetMetricSpecForScaling(context.Context) []v2.MetricSpec {
```

Comment:
Following the general convention, `s` shall be used as the receiver variable for `*neo4jScaler`. Please update in other places too.

```suggestion
func (s *neo4jScaler) GetMetricSpecForScaling(context.Context) []v2.MetricSpec {
```

Reply:
Sure